Source/DafnyCore/Verifier/BoogieGenerator.Types.cs

```diff
@@ -1524,8 +1525,8 @@ void AddWellformednessCheck(RedirectingTypeDecl decl) {
       } else {
         // before trying 0 as a witness, check that 0 can be assigned to baseType
         witnessString = Printer.ExprToString(options, witness);
-        CheckResultToBeInType(decl.tok, witness, decl.Var.Type, locals, witnessCheckBuilder, etran, $"trying witness {witnessString}: ");
-        witnessExpr = Substitute(decl.Constraint, decl.Var, witness);
+        CheckResultToBeInType(decl.tok, witness, baseType, locals, witnessCheckBuilder, etran, $"trying witness {witnessString}: ");
+        witnessExpr = decl.Constraint != null ? Substitute(decl.Constraint, decl.Var, witness) : null;
```

Comment:
Same question as above. I'm surprised that witnessExpr can be null even if there is a witness given.